Orange/widgets/utils/plot/owplotgui.py

```diff
@@ -51,165 +53,175 @@
+            txt = [" Add ", " Remove "][is_selected]
+            txtw = painter.fontMetrics().width(txt)
+            painter.save()
+            painter.setPen(Qt.NoPen)
+            painter.setBrush(QColor("#ccc"))
+            brect = QRect(rect.x() + rect.width() - 8 - txtw, rect.y(),
+                          txtw, rect.height())
+            painter.drawRoundedRect(brect, 4, 4)
+            painter.restore()
+            painter.drawText(brect, Qt.AlignCenter, txt)
+
+        painter.save()
+        double_pen = painter.pen()
+        double_pen.setWidth(2 * double_pen.width())
+        if is_selected:
+            next = index.sibling(index.row() + 1, index.column())
+            if not next.isValid():
+                painter.setPen(double_pen)
+                painter.drawLine(rect.bottomLeft(), rect.bottomRight())
+            elif not next.data(VariableSelectionModel.IsSelected):
+                painter.drawLine(rect.bottomLeft(), rect.bottomRight())
+        elif not index.row():
+            down = QPoint(0, painter.pen().width())
+            painter.setPen(double_pen)
+            painter.drawLine(rect.topLeft() + down, rect.topRight() + down)
+        else:
+            prev = index.sibling(index.row() - 1, index.column())
+            if prev.data(VariableSelectionModel.IsSelected):
+                painter.drawLine(rect.topLeft(), rect.topRight())
+        painter.restore()
+
+        super().paint(painter, option, index)
+
+
+class VariableSelectionView(QListView):
+    dragDropActionDidComplete = pyqtSignal(int)
+
+    def __init__(self, *args, acceptedType=None, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.setSizePolicy(
+            QSizePolicy(QSizePolicy.Minimum, QSizePolicy.MinimumExpanding))
+        self.setMinimumHeight(10)
+        self.setMouseTracking(True)
+        self.setAttribute(Qt.WA_Hover)
+
+        self.setSelectionMode(self.SingleSelection)
+        self.setAutoScroll(False)  # Prevent scrolling to removed item
+        self.setDragEnabled(True)
+        self.setDropIndicatorShown(True)
+        self.setDragDropMode(self.InternalMove)
+        self.setDefaultDropAction(Qt.MoveAction)
+        self.setDragDropOverwriteMode(False)
+        self.setUniformItemSizes(True)
+
+        self.setItemDelegate(VariablesDelegate())
+
+    def sizeHint(self):
+        return QSize(1, 50)
+
+    def mouseMoveEvent(self, e):
+        super().mouseMoveEvent(e)
+        self.update()
+
+    def leaveEvent(self, e):
+        super().leaveEvent(e)
+        self.update()  # BUG: This update has no effect, at least not on macOs
+
+    def startDrag(self, supportedActions):
+        super().startDrag(supportedActions)
+        self.selectionModel().clearSelection()
+
+
+def variables_selection(widget, master, model):
+    def update_list():
+        proxy.sort(0)
+        proxy.invalidate()
+        view.selectionModel().clearSelection()
+
+    filter, view = variables_filter(
```

Comment:
`filter` is not my favourite choice of a variable name since it is a built-in function. How about `filter_edit` as in the `variables_filter`?

Reply:
Mine neither. Pylint's neither. Changed to `filter_edit`.